code/modules/SCP/scp017.dm

```diff
@@ -0,0 +1,637 @@
+	if(current_ai_state == VINDICTIVE)
+		var/mob/living/target_mob = get_highest_priority_target()
+		if(target_mob == the_target) // If it's the high priority target in vindictive state
+			return TRUE // Attack regardless of light
+		// Also prioritize breaking lights
+		if(istype(the_target, /obj/machinery/light) && (the_target:on))
+			return TRUE
+
+	if((Tturf.get_lumcount() > shadow_threshold) || (!Tarea.area_lighting == AREA_LIGHTING_DYNAMIC))
+		return FALSE
+	return TRUE
+
+/mob/living/simple_animal/hostile/scp017/Move(turf/newloc, safety = TRUE, ignore_light_check = FALSE)
+	var/turf/Tturf = newloc //Cleanliness
+	var/area/Tarea = get_area(newloc)
+	if(!ignore_light_check && ((Tturf.get_lumcount() > shadow_threshold) || (!Tarea.area_lighting == AREA_LIGHTING_DYNAMIC)))
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/scp017/gib()
+	return FALSE
+
+/mob/living/simple_animal/hostile/scp017/dust(just_ash, drop_items, force)
+	return FALSE
+
+
+//Death
+
+/mob/living/simple_animal/hostile/scp017/death(gibbed, cause_of_death = "disappears in a puff of smoke")
+	. = ..()
+	var/turf/T = get_turf(src)
+
+	do_smoke(amount = 1, location = T)
+
+	ghostize()
+	qdel(src)
+
+/mob/living/simple_animal/hostile/scp017/proc/ShadowVeil()
+	if(world.time < last_shadow_veil_time + shadow_veil_cooldown)
+		return FALSE
+
+	last_shadow_veil_time = world.time
+	SafeVisibleMessage(span_danger("[src] expands, casting a deep shadow around itself!"))
+
+	var/list/affected_turfs = list()
+	for(var/turf/T in range(3, src)) // 3x3 radius for now
+		if(isturf(T))
+			var/area/A = get_area(T)
+			if(A.area_lighting == AREA_LIGHTING_DYNAMIC) // Only affect dynamic lighting areas
+				T.set_light(0, 0, 0)
+				affected_turfs += T
+
+	ADD_TRAIT(src, TRAIT_IMMOBILIZED, "SCP017_SHADOW_VEIL_IMMUNITY")
+	addtimer(CALLBACK(src, PROC_REF(RemoveShadowVeilImmunity)), shadow_veil_duration)
+
+	return TRUE
+
+/mob/living/simple_animal/hostile/scp017/proc/RemoveShadowVeilImmunity()
+	if(HAS_TRAIT(src, TRAIT_IMMOBILIZED))
+		REMOVE_TRAIT(src, TRAIT_IMMOBILIZED, "SCP017_SHADOW_VEIL_IMMUNITY")
+		SafeVisibleMessage(span_notice("The shadows around [src] recede."))
+
+/mob/living/simple_animal/hostile/scp017/proc/IsStrongLightSource(atom/A)
+	if(istype(A, /obj/machinery/light))
+		var/obj/machinery/light/L = A
+		if(L.on && L.light_power > 0)
+			return TRUE
+	if(istype(A, /obj/item/flashlight))
+		var/obj/item/flashlight/F = A
+		if(F.on && F.light_power > 0)
+			return TRUE
+	return FALSE
+
+/mob/living/simple_animal/hostile/scp017/proc/mob_has_strong_light_source(mob/living/M)
+	if(!M)
+		return FALSE
+	// Check if the mob itself is a light source (e.g., a glowing creature)
+	if(IsStrongLightSource(M))
+		return TRUE
+	// Check mob's held items
+	if(M.held_items)
+		for(var/obj/item/I in M.held_items)
+			if(IsStrongLightSource(I))
+				return TRUE
+	// Check mob's inventory
+	if(M.contents)
+		for(var/obj/item/I in M.contents)
```

Comment:
Not safe, this includes organs and bodyparts. Consider `M.get_all_contents() - M.organs - M.bodyparts` if you REALLY want to check *everything*